test/updater_unittest.cc

```diff
@@ -215,6 +216,73 @@ TEST_F(UpdaterTest, UpdateResourceCallback) {
             store.LookupResource("test-prefix"));
 }
 
+bool WaitForResource(const MetadataStore& store,
+                     const MonitoredResource& resource) {
+  for (int i = 0; i < 30; i++) {
+    const auto metadata_map = store.GetMetadataMap();
+    if (metadata_map.find(resource) != metadata_map.end()) {
+      return true;
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  }
+  return false;
+}
+
+// PollingMetadataUpdater that uses a FakeClock.
+class FakePollingMetadataUpdater : public PollingMetadataUpdater {
+ public:
+  FakePollingMetadataUpdater(
+      const Configuration& config, MetadataStore* store,
+      const std::string& name, int period_s,
+      std::function<std::vector<ResourceMetadata>()> query_metadata)
+      : PollingMetadataUpdater(
+          config, store, name, period_s, query_metadata,
+          std::unique_ptr<Timer>(new TimerImpl<testing::FakeClock>(
+              config.VerboseLogging(), name))) {}
```

Comment:
Since there's no way to set `config.VerboseLogging()` to `true` in a test (unless you have a friend class), this might as well just be `false`.